static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -57,36 +127,115 @@ function SelectProjectStep({
   Body: ModalBody,
   Header: ModalHeader,
   Footer: ModalFooter,
+  closeModal,
   toStep,
   step,
+  project,
+  setProject,
 }: OnboardingStepProps) {
-  const onNext = useCallback(
-    (platform: 'iOS' | 'Android') => {
+  const {projects} = useProjects();
+
+  const onFormSubmit = useCallback(
+    (evt: React.FormEvent) => {
+      evt.preventDefault();
+
+      if (!project?.platform) {
+        return;
+      }
+
+      const nextStep = platformToInstructionsMapping[project.platform];
+      if (nextStep === undefined) {
+        // We should probably be throwing here
+        return;
+      }
+
       toStep({
         previous: step,
-        current:
-          platform === 'Android'
-            ? AndroidSendDebugFilesInstruction
-            : IOSSendDebugFilesInstruction,
+        current: nextStep,
         next: null,
       });
     },
-    [step, toStep]
+    [project, step, toStep]
+  );
+
+  const onProjectSelect = useCallback(
+    (value: Project) => {
+      setProject(value);
+    },
+    [setProject]
```

Comment:
is there a point to this `useCallback`? it's just wrapping `setProject` without adding any functionality to it

Reply:
Oh yea, we dont need this. I had the logic a bit differently setup at first and it stuck around